src/fuzzfetch/fetch.py

```diff
@@ -617,25 +620,32 @@ def extract_zip(self, suffix, path='.'):
         finally:
             os.unlink(zip_fn)
 
-    def extract_tar(self, path='.'):
+    def extract_tar(self, suffix, path='.'):
         """
-        Extract builds with .tar.bz2 extension
-        Only extracts the top-level directory "firefox"
+        Extract builds with .tar.(bz2|gz) extension
+        When unpacking a build archive, only extract the firefox directory
+
+        @type suffix:
+        @param suffix:
 
         @type path:
         @param path:
         """
-        tar_fd, tar_fn = tempfile.mkstemp(prefix='fuzzfetch-', suffix='.tar.bz2')
+        mode = 'bz2' if suffix.endswith('tar.bz2') else 'gz'
```

Comment:
This doesn't look right. We should check that the input is either 'tar.bz2' or 'tar.gz'.